### PR TITLE
Fix python 3.11 tests, Expand test matrix to 3.8-3.12

### PR DESCRIPTION
**Pull request by crazygao**
# Description

python 3.11 introduce some new features
num.type == ValueType.int  (old ones are num)

python 3.11 backtrace add a tick indicating where is ZeroDivision. Like the following.

1 / 0
~~^~~

So regex match may fail on 3.11

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/tests/executor/unittests/_utils/test_exception_utils.py [hunk 1]
```diff
@@ -481,6 +481,7 @@ def test_tool_execution_error(self):
             r"Traceback \(most recent call last\):\n"
             r'  File ".*test_exception_utils.py", line .*, in code_with_bug\n'
             r"    1 / 0\n"
+            r"(.*\n)?"
```
Comment (thy09) [inline src/promptflow/tests/executor/unittests/_utils/test_exception_utils.py:484]: Add some comments here?

Comment (crazygao) [reply src/promptflow/tests/executor/unittests/_utils/test_exception_utils.py:484]: Okay.

File: .github/workflows/promptflow-release-testing-matrix.yml [hunk 2]
```diff
@@ -105,7 +105,7 @@ jobs:
       matrix:
         os: [windows-latest]
         testType: [sdk-cli, executor-e2e, executor-unit]
-        pythonVersion: ['3.9', '3.10', '3.11']
+        pythonVersion: ['3.8', '3.9', '3.10', '3.11', '3.12']
```
Comment (D-W-) [inline .github/workflows/promptflow-release-testing-matrix.yml:108]: No need to cover 3.12?
image

**Review by D-W- — APPROVED** [review]

File: .github/workflows/promptflow-release-testing-matrix.yml (hunk 2 above)
Comment (crazygao) [reply .github/workflows/promptflow-release-testing-matrix.yml:108]: Not a blocking pipeline

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**